### PR TITLE
Add grouping option for date formatting.

### PR DESCRIPTION
**Pull request by cjurden**

### CODE REVIEW

File: js/Source/Charts/Chart.js [hunk 1]
```diff
@@ -507,18 +508,20 @@ Chart = new Class({
           formatter: function(){
             if (self.options.xAxis.type == 'datetime'){
               // parse the date
-              var d = new Date(this.value);
-              switch(self.grouping){
+              var offset = new Date().getTimezoneOffset();
```
Comment (drn) [inline js/Source/Charts/Chart.js]: we can remove these offsets

File: js/Source/Charts/Chart.js [hunk 2]
```diff
@@ -615,7 +618,7 @@ Chart = new Class({
           // x position of our line is the chart plotLeft + the offset of the point
           // and then we subtract half of the label's width to center it and subtract
           // 1 more to center on our 2px wide crosshair line
-          var x = (point.plotX + this.chart.plotLeft - (labelWidth/2) - 1);
+          var x = (point.plotX + this.chart.plotLeft - (labelWidth/2) - 2);
```
Comment (drn) [inline js/Source/Charts/Chart.js]: do we need this change here?

File: js/Source/Charts/Chart.js [hunk 3]
```diff
@@ -279,6 +279,7 @@ Chart = new Class({
     var v2yAxis = options.yAxis;
     var size = this.getSizeOptions();
     var self = this;
+    if (options.columnGrouping) this.options.columnGrouping = options.columnGrouping;
```
Comment (anutron) [inline js/Source/Charts/Chart.js]: Just confirming, this allows you to specify column grouping in the response (in addition to the options passed in at instantiation), yes? That works.

File: js/Source/Charts/Chart.js [hunk 4]
```diff
@@ -507,18 +508,17 @@ Chart = new Class({
...
                 case 'week':
                   return "<span class='axis-date'><span class='week'>week of</span><span class='sub'>" +
-                          d.format("%b %e") + "</span></span>";
+                          Highcharts.dateFormat("%b %e", this.value) + "</span></span>";
                 default:
                   // format it; it gets styled by css
-                  return "<span class='axis-date'><span class='day'>" + d.format("%a")[0] +
-                         "</span><span class='sub'>" +  d.format("%b %e") + "</span></span>";
+                  return "<span class='axis-date'><span class='day'>" + Highcharts.dateFormat("%a", this.value)[0] +
+                         "</span><span class='sub'>" +  Highcharts.dateFormat("%b %e", this.value) + "</span></span>";
```
Comment (anutron) [inline js/Source/Charts/Chart.js]: GREAT catch.

**Comment by anutron** [comment]
@cjurden  be sure to do all the homework here with this commit. Before you merge to master, run `grunt production` and include the artifacts in the commit.

**Comment by cjurden** [comment]
@drn think I made those changes in the compiled project's js by mistake...

**Comment by drn** [comment]
@cjurden LGTM - let's get this out and into prod

**Pull request merged**